docs/Profiles.md

```diff
@@ -0,0 +1,259 @@
+and boost adoption of the Lottie animation format.
+
+
+# Objective
+
+Introduce the notion of, and propose a tentative/initial set of Modules and
+Profiles.
+
+
+# Modules
+
+A ***module*** is a concise functional part of the general feature space, a
+group of related features.
+
+Inspired by the [CSS standardization process](
+https://www.w3.org/Style/2011/CSS-process.en.html#:~:text=MODULES%20AND%20SNAPSHOTS
+), modules can be versioned independently, thus allowing the specification to
+evolve organically instead of a single/monolithic document. They also allow a
+more concise formulation of profiles, by referring to modules/groups instead of
+individual features.
+
+The level of granularity is an open question: at one extreme, one could
+potentially consider each feature its own module – but that would diminish the
+value or outright defeat the purpose of using modules at all; at the other
+extreme, modules that are too coarse would make it difficult to accurately
+describe existing players’ feature sets.
+
+A middle-of-the road approach will likely work best in the long run.  E.g.
+
+* Values and Keyframes: (value types, keyframe format, interpolation options)
+* Transforms: 2D transforms, 3D transforms?
+* Basic layers: null, solid, image, precomp
+* Shape layers
+* Text layers
+* Fonts
+* Advanced layers: camera, audio
+* Basic effects: (common set of effects that can be supported in most players)
+* Advanced effects: (fancy effects that may require e.g. custom shaders)
+
+
+# Profiles
+
+A ***profile*** is a group of supported limits, features, and formats that
+specifies a set of Lottie player capabilities.
+
+Profiles are useful for describing feature support across the heterogeneous
+Lottie ecosystem, and for offering compatibility guarantees – e.g. *“LottieWeb
+v1.2 supports the Core Profile v1.0 plus the following extensions: …”*.
+
+Profiles can be defined in terms of individual features, or by referencing
+existing modules or other profiles.  For the actual format, both a
+human-readable version and a machine-parsable version are useful: the
+human-readable version is informative for animation authors, helping them
+constrain the design to the features available in the target profile, while
+the machine-targeted version can be used for tooling (e.g. validation at export
+time and at run time).
+
+The Lottie community should offer a set of official profiles, with wide
+applicability – e.g.
+
+* **Baseline Profile** - A minimum set of features that should be supported by
+                         any Lottie implementation.  Anything less cannot claim
+                          Lottie compatibility.
+* **Core Profile** - Representative, commonly supported features across
+                     existing players.  Someone targeting the Core profile can
+                     expect their assets to be supported on all major engines.
+* **Embedded Profile**? - A reduced set of features that can be supported on
```

Comment:
Would Embedded profile be different from baseline?

Also, IMO the "Baseline" / "Core" distinction is a bit ambiguous, perhaps we can rename them?

Reply:
+1 on addressing ambiguity on core and baseline.